src/BaseResponse.php

```diff
@@ -235,10 +235,19 @@ public function send() {
 		}
 
 		// messages cookie
-		if( isset($this->messages) ) {
+		if( isset($this->messages) && !empty($this->messages) ) {
 			setcookie('YOLK_MESSAGES', base64_encode(serialize($this->messages)), 0, '/');
 			unset($this->cookies['YOLK_MESSAGES']);
 		}
+		else {
+			$existing = array_key_exists('YOLK_MESSAGES', $_COOKIE) 
+				? $_COOKIE['YOLK_MESSAGES'] : null;
+			// if the YOLK_MESSAGES cookie is a serialized empty array, we should unset it
+			$existing_cookie_empty = $existing == "YTowOnt9";
```

Comment:
perhaps compare to a real b64 enced serialised array *just in case* php change how they serialise?